src/main/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtils.java

```diff
@@ -0,0 +1,116 @@
+package bio.terra.service.resourcemanagement;
+
+import bio.terra.common.Table;
+import bio.terra.model.AccessInfoBigQueryModel;
+import bio.terra.model.AccessInfoBigQueryModelTable;
+import bio.terra.model.AccessInfoModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.stringtemplate.v4.ST;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utilities for building strings to access metadata
+ */
+public final class MetadataDataAccessUtils {
+    private static final String BIGQUERY_DATASET_LINK = "https://console.cloud.google.com/bigquery?project=<project>&" +
+        "ws=!<dataset>&d=<dataset>&p=<project>&page=<page>";
+    private static final String BIGQUERY_TABLE_LINK = BIGQUERY_DATASET_LINK + "&t=<table>";
+    private static final String BIGQUERY_TABLE_ADDRESS = "<project>.<dataset>.<table>";
+    private static final String BIGQUERY_DATASET_ID = "<project>:<dataset>";
+    private static final String BIGQUERY_TABLE_ID = "<dataset_id>.<table>";
+    private static final String BIGQUERY_BASE_QUERY = "SELECT * FROM `<table_address>` LIMIT 1000";
+
+    private MetadataDataAccessUtils() {
+    }
+
+    /**
+     * Nature of the page to link to in the BigQuery UI
+     */
+    enum LinkPage {
+        DATASET("dataset"),
+        TABLE("table");
+        private final String value;
+        LinkPage(final String value) {
+            this.value = value;
+        }
+    }
+
+    /**
+     * Generate an {@link AccessInfoModel} from a Snapshot
+     */
+    public static AccessInfoModel accessInfoFromSnapshot(final Snapshot snapshot) {
+        return makAccessInfo(
+            snapshot.getName(),
+            snapshot.getProjectResource().getGoogleProjectId(),
+            snapshot.getTables()
+        );
+    }
+
+    /**
+     * Generate an {@link AccessInfoModel} from a Dataset
+     */
+    public static AccessInfoModel accessInfoFromDataset(final Dataset dataset) {
+        return makAccessInfo(
+            BigQueryPdao.prefixName(dataset.getName()),
+            dataset.getProjectResource().getGoogleProjectId(),
+            dataset.getTables()
+        );
+    }
+
+    private static AccessInfoModel makAccessInfo(
+        final String bqDatasetName,
+        final String googleProjectId,
+        final List<? extends Table> tables
+    ) {
```

Comment:
Also not typical formatting
```suggestion
        final List<? extends Table> tables) {
```